core/validation/failures.py

```diff
@@ -273,15 +274,15 @@ class InvalidEnumValueFailure(ValidationFailure):
 
     sheet: str
     column: str
-    row: int
+    row: list[int]
     row_values: tuple
     value: Any
 
     def __str__(self):
         """Method to get the string representation of the invalid enum value failure."""
         return (
             f'Enum Value Failure: Sheet "{self.sheet}" Column "{self.column}" '
-            f"Row {self.row + 2} "  # +2 for Excel 1-index and table header row
+            f"Row {self.row[0] + 2} "  # +2 for Excel 1-index and table header row
```

Comment:
We don't have to do it everywhere else - apart from here (because this is a legacy message and doesn't follow our to_message() protocol), and in our tests (which will have to happen anyway as even incrementing indexes +2 at the end of transformation means testing validation functions require us to mimic the +2). We have a function that constructs the cell locations (from the R4-specific PR) from the index and sheet/col, and the incrementation happens in there, but either way that function has to be called in every to_message() call.

We can also remove it from that function and put it at the end of the R3 and R4 transformation pipelines, which might be more readable from a developer perspective. I'll rebase this branch onto main now that the R4-specific PR has been merged and refactor everything now that it is all in the same branch.